### PR TITLE
Move most of utils out into separate crates

### PR DESCRIPTION
**Pull request by DieracDelta**

### CODE REVIEW

**Review by Ancient123 — DISMISSED** [review]

**Comment by Ancient123** [comment]
> [CLA assistant check](https://cla-assistant.io/EspressoSystems/HotShot?pullRequest=784) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/HotShot?pullRequest=784) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/HotShot?pullRequest=784) it.

I think CLA bot is bugging out. Because it should bypass for people in the EspressoSystems org.

**Comment by DieracDelta** [comment]
Looks like CLA is fixed. CI should pass now, so we can probably merge this.

File: libp2p-networking/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # USAGE
 
-Networking library inteded for use with HotShot. Builds upon abstractions on libp2p-rs.
+Networking library inteded for use with HotShot. Builds upon compatibilitys on libp2p-rs.
```
Comment (shenkeyao) [inline libp2p-networking/README.md:3]: Typo, compatibilities or compatibility?

**Review by shenkeyao — CHANGES_REQUESTED** [review]
LGTM except for a typo.

**Review by shenkeyao — APPROVED** [review]

**Pull request merged**